File: validation/poc-00/java-candidate/src/main/java/dev/vra/poc00/domain/InventoryReservation.java
package dev.vra.poc00.domain;

import java.util.Objects;
import java.util.UUID;

/** Representative value only: this POC does not persist reservation identities. */
public record InventoryReservation(UUID id, SkuId skuId, int quantity, State state) {
    public enum State { ACTIVE, RELEASED }
    public InventoryReservation {
        Objects.requireNonNull(id, "id");
        Objects.requireNonNull(skuId, "skuId");
        Objects.requireNonNull(state, "state");
        InventoryBalance.requireQuantity(quantity);
    }
}

File: validation/poc-00/kotlin-candidate/src/main/kotlin/dev/vra/poc00/kotlin/domain/InventoryReservation.kt
package dev.vra.poc00.kotlin.domain

import java.util.UUID

data class InventoryReservation(
    val id: UUID,
    val skuId: SkuId,
    val quantity: Int,
    val state: State
) {
    enum class State { ACTIVE, RELEASED }

    init {
        InventoryBalance.requireQuantity(quantity)
    }
}
